packages/vscode-extension/src/webview/components/UrlSelect.tsx

```diff
@@ -168,24 +222,32 @@ function UrlSelect({ onValueChange, navigationHistory, disabled, dropdownOnly }:
                 setTimeout(() => textfieldRef.current?.focus(), 0);
               } else if (dropdownOnly) {
                 setIsDropdownOpen(!isDropdownOpen);
-                textfieldRef.current?.blur();
+                setTimeout(() => textfieldRef.current?.blur(), 0);
               }
             }}
             onKeyDown={(e) => {
+              if (dynamicSegmentNames.length > 0 && e.key === "Tab") {
+                if (e.shiftKey && currentDynamicSegment > 0) {
+                  e.preventDefault();
+                  setCurrentDynamicSegment(currentDynamicSegment - 1);
+                } else if (!e.shiftKey && currentDynamicSegment < dynamicSegmentNames.length - 1) {
+                  e.preventDefault();
+                  setCurrentDynamicSegment(currentDynamicSegment + 1);
+                }
+              }
               if (e.key === "Enter") {
                 closeDropdownWithValue(textfieldRef.current?.value ?? "");
-                textfieldRef.current?.blur();
               }
               if (e.key === "Escape") {
                 setIsDropdownOpen(false);
-                textfieldRef.current?.blur();
+                setTimeout(() => textfieldRef.current?.blur(), 0);
               }
               if (e.key === "ArrowDown") {
                 if (isDropdownOpen) {
                   navigateBetweenItems(
                     e,
                     undefined,
-                    document.querySelector(".url-select-item") as HTMLDivElement
+                    itemsRef.current[0]?.current as UrlSelectFocusable
```

Comment:
we call `navigateBetweenItems` here but in this context word "navigate" means that we update the navigation state of the app. I understand that this method in that context doesn't actually "navigate". Can we change the name of it in that case as it is misleading.